src/nvme/tree.c

```diff
@@ -1374,9 +1374,11 @@ nvme_ctrl_t nvme_scan_ctrl(nvme_root_t r, const char *name)
 	free(subsysnqn);
 
 	ret = 0;
-	if (s && !s->name && subsysname)
-		ret = nvme_init_subsystem(s, subsysname);
-	free(subsysname);
+	if (subsysname) {
+		if (s && !s->name)
+			ret = nvme_init_subsystem(s, subsysname);
+		free(subsysname);
```

Comment:
Oh, I lived in assumption that `free()` called on NULL pointer does nothing. At least the manpage says so but I was always wondering if all the libcs out there follow this semantic.
(FWIW, it's me to blame for this `subsysname` free here)

Reply:
IIRC free(NULL) is a no op for C99

Reply:
Ok, ok. I've dropped that patch now.